## 5. 开店潜力指数可视化（05_plot_potential_index.ipynb）
### 任务目标
基于数据清洗后生成的行政区统计数据，绘制广州市各行政区便利店开店潜力指数柱状图，直观呈现不同区域的扩张空间差异，为选址优先级提供量化支撑。

### 输入数据
- `data/district_stats.csv`：包含各行政区门店数、常住人口、万人密度、开店潜力指数的统计结果

### 核心处理步骤
1.  读取行政区统计数据，按开店潜力指数从高到低排序
2.  绘制柱状图，标注各行政区的真实潜力指数数值
3.  优化图表标注、配色与排版，明确指数含义（数值越高，扩张空间越大）
4.  输出高分辨率PNG图表，供汇报使用

### 输出文件
- `output/fig3_potential.png`：广州市各行政区便利店开店潜力指数分布图

In [2]:
import pandas as pd
import matplotlib.pyplot as plt
import os
import numpy as np

# 创建输出文件夹
os.makedirs("output", exist_ok=True)

# 读取真实行政区统计数据，自动过滤空值
df = pd.read_csv("data/district_stats.csv")
# 核心修复：删除包含NaN的无效行，避免空值导致报错
df = df.dropna(subset=["district", "potential_index"]).copy()
# 按潜力指数降序排序
df = df.sort_values("potential_index", ascending=False).reset_index(drop=True)

# 学术风格配置
plt.rcParams["font.sans-serif"] = ["SimHei", "Microsoft YaHei", "Times New Roman"]
plt.rcParams["axes.unicode_minus"] = False
plt.rcParams["figure.facecolor"] = "white"

# 绘制潜力指数柱状图
fig, ax = plt.subplots(figsize=(12, 7))
bars = ax.bar(
    df["district"], 
    df["potential_index"], 
    color="#C73E1D",  # 学术红，突出潜力差异
    alpha=0.8,
    width=0.6
)

# 核心修复：安全处理数值标签，兼容NaN/空值
for bar in bars:
    height = bar.get_height()
    # 跳过空值/无效值，避免报错
    if not np.isnan(height):
        ax.text(
            bar.get_x() + bar.get_width()/2,
            height + 50,  # 标签偏移
            f"{int(round(height))}",  # 先四舍五入再转整数，避免浮点精度问题
            ha="center", 
            va="bottom", 
            fontsize=11,
            fontweight="bold"
        )

# 图表标注
ax.set_title("广州市各行政区便利店开店潜力指数", fontsize=16, fontweight="bold", pad=20)
ax.set_xlabel("行政区", fontsize=13, fontweight="bold")
ax.set_ylabel("开店潜力指数（数值越高，扩张空间越大）", fontsize=13, fontweight="bold")
ax.set_ylim(0, max(df["potential_index"]) * 1.15)
ax.grid(axis="y", alpha=0.3, linestyle="-")

# 调整布局并保存
plt.tight_layout()
plt.savefig("output/fig3_potential.png", dpi=300, bbox_inches="tight", facecolor="white")
plt.close()

# 输出核对日志
print("✅ 开店潜力图生成完成：output/fig3_potential.png")
print("\n=== 开店潜力指数统计（按潜力降序）===")
print(df[["district", "density_per_10k", "potential_index"]].round(2))

✅ 开店潜力图生成完成：output/fig3_potential.png

=== 开店潜力指数统计（按潜力降序）===
  district  density_per_10k  potential_index
0      白云区             0.76           4145.8
1      番禺区             0.98           3104.6
2      天河区             1.28           2403.2
3      花都区             0.75           1919.2
4      海珠区             1.16           1888.6
5      黄埔区             1.01           1398.6
6      越秀区             1.40           1246.0
7      荔湾区             1.08           1056.6


## 开店潜力指数结论（基于可视化结果）
1.  白云区潜力指数最高，达到4146，是全市扩张空间最大的区域，为选址第一优先级。
2.  番禺区潜力指数3105，位居全市第二，人口基数大+现有密度偏低，扩张空间充足。
3.  天河区、花都区、海珠区潜力指数分别为2403、1919、1889，处于中等水平，适合稳步布局。
4.  黄埔区、越秀区、荔湾区潜力指数分别为1399、1246、1057，处于全市较低水平，市场相对饱和，新增门店需谨慎评估。
5.  整体来看，全市便利店扩张潜力呈现「远郊＞近郊＞核心城区」的特征，人口基数大的区域扩张空间显著更高。